runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WriteFilesTranslation.java

```diff
@@ -58,12 +63,17 @@
 
   @VisibleForTesting
   static WriteFilesPayload toProto(WriteFiles<?, ?, ?> transform) {
+    Map<String, SideInput> sideInputs = Maps.newHashMap();
+    for (PCollectionView<?> view : transform.getSink().getDynamicDestinations().getSideInputs()) {
+      sideInputs.put(view.getTagInternal().getId(), ParDoTranslation.toProto(view));
+    }
     return WriteFilesPayload.newBuilder()
         .setSink(toProto(transform.getSink()))
         .setFormatFunction(toProto(transform.getFormatFunction()))
         .setWindowedWrites(transform.isWindowedWrites())
         .setRunnerDeterminedSharding(
             transform.getNumShards() == null && transform.getSharding() == null)
+        .putAllDynamicDestinationSideInputs(sideInputs)
```

Comment:
Are the PCollections referenced here external to the WriteFiles composite? It seems they should be accessed not by way of the sink but by way of WriteFiles.getDynDestSideInputs() or maybe just WriteFiles.getAdditionalInputs.